Tools/CISupport/Shared/de-duper

```diff
@@ -114,23 +119,37 @@ def main(args):
             if os.stat(full_path).st_nlink > 1:
                 continue
             sha_value = sha1(full_path)
-            if sha_value in existing_links:
+            sha_path = os.path.join(args.storage, f'{sha_value}.png')
+            assert sha_path != full_path
+            if os.path.isfile(sha_path):
                 if args.dry_run:
                     print(f'[dry-run] Linking {full_path} to {sha_value}.png')
                 else:
-                    os.remove(full_path)
-                    os.link(os.path.join(args.storage, f'{sha_value}.png'), full_path)
+                    try:
+                        tempfile = os.path.join(root, f'{sha_value}.png')
+                        os.link(sha_path, tempfile)
+                        os.replace(tempfile, full_path)
+                    except OSError as e:
+                        if e.errno == errno.EMLINK:
+                            # sha_path already has LINK_MAX links, so create a copy of that file at
+                            # full_path, and create a new hard link within storage pointing at another
+                            # identical file. In practice, the previous network of hard-linked files will
+                            # slowly be deleted as old results are pruned.
+                            os.remove(sha_path)
+                            os.link(full_path, sha_path)
```

Comment:
Nit: we should maybe call `os.replace` here, or comment why not.

(I think in practice, because `sha_path` is never actually directly referenced, that it simply doesn't matter if we end up in a state where that file is deleted but not recreated?)